compiler/src/org.graalvm.compiler.truffle.compiler.benchmark/src/org/graalvm/compiler/truffle/compiler/benchmark/pelang/PELangIfNode.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2018, 2018, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.graalvm.compiler.truffle.compiler.benchmark.pelang;
+
+import com.oracle.truffle.api.frame.VirtualFrame;
+
+public final class PELangIfNode extends PELangExpressionNode {
+
+    @Child private PELangExpressionNode conditionNode;
+    @Child private PELangExpressionNode thenNode;
+    @Child private PELangExpressionNode elseNode;
+
+    public PELangIfNode(PELangExpressionNode conditionNode, PELangExpressionNode thenNode, PELangExpressionNode elseNode) {
+        this.conditionNode = conditionNode;
+        this.thenNode = thenNode;
+        this.elseNode = elseNode;
+    }
+
+    @Override
+    public Object executeGeneric(VirtualFrame frame) {
+        long conditionResult = conditionNode.evaluateCondition(frame);
+
+        if (conditionResult == 0L) {
```

Comment:
Normal languages would have a counting condition profile here. Not sure if you want to be close to languages for this.